docs/src/modules/spring/pages/access-control.adoc

```diff
@@ -1 +1,212 @@
 = Access Control Lists (ACLs)
+
+Access Control Lists (ACLs) allow you to control which services and sources of requests may access your services.
+
+Within a Kalix project, all communication between services uses Mutual TLS (mTLS). This is injected transparently by Kalix. The clients you use to make calls from one service to another do NOT need to be configured to do this. Kalix transparently captures outgoing requests to other services and wraps them in an mTLS connection.
+
+Based on this mTLS support, Kalix is able to read and apply policies based on where the request was made from.
+
+== Principals
+
+A principal in Kalix is an abstract concept that represents anything that can make or be the source of a request. Principals that are currently supported by Kalix include other services, and the internet. Kalix uses the above described mTLS support to associate requests with one or more principals.
+
+Note that requests that have the internet principal are requests that Kalix has identified as coming through the Kalix ingress, according to a configured route. This is identified by mTLS, however it does not imply that mTLS has been used to connect to the ingress from the client in the internet. These are separate hops. To configure mTLS from internet clients, see https://docs.kalix.io/services/using-client-certificates.html[Client certificates]
+
+== Configuring ACLs
+
+Kalix ACLs consist of two lists of principal matchers. One to allow to invoke a method, and the other to deny to invoke a method. For a request to be allowed, at least one principal associated with a request must be matched by at least one principal matcher in the allow list, and no principals associated with the request may match any principal matchers in the deny list.
+
+Here is an example ACL on a method:
+
+```java
+    @PostMapping
+    @Acl(
+       allow = @Acl.Matcher(service = "*"), 
+       deny =  @Acl.Matcher(service = "my-service"))
+    public Effect<String> createUser(@RequestBody CreateEmployee create) {
+     ...
+    }
+```
```

Comment:
We should have all those snippets in a sample so we can prove that they are compiling correctly.

Isn't the gRPC one using snippets?

Reply:
gRPC is not using snippets.

Reply:
I'll add them np